src/pages/MainPage.jsx

```diff
@@ -28,43 +29,53 @@ const photos = [
   'https://images.unsplash.com/photo-1617505907947-9cb31eb80183?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1867&q=80',
 ];
 
+let user;
+
 function MainPage() {
+  async function getUser(token, id) {
+    try {
+      API.getInfo(token, id)
+        .then((res) => {
+          user = res;
+        });
+    } catch (e) {
+      // need to handle this error state.
```

Comment:
In this case, I'm pretty sure we can just do `history.replace('/')` (using replace here so that the back button doesn't take you back to this page, might have to come back to this later)

Reply:
Ill do that for now but if we have time later I think we should make an error page or component